src/library/analysisfeature.cpp

```diff
@@ -112,88 +111,74 @@ void AnalysisFeature::activate() {
     emit(enableCoverArtDisplay(true));
 }
 
-namespace {
-    inline
-    AnalyzerQueue::Mode getAnalyzerQueueMode(
-            const UserSettingsPointer& pConfig) {
-        if (pConfig->getValue<bool>(ConfigKey("[Library]", "EnableWaveformGenerationWithAnalysis"), true)) {
-            return AnalyzerQueue::Mode::Default;
-        } else {
-            return AnalyzerQueue::Mode::WithoutWaveform;
-        }
-    }
-} // anonymous namespace
-
 void AnalysisFeature::analyzeTracks(QList<TrackId> trackIds) {
-    if (m_pAnalyzerQueue == NULL) {
+    if (!m_pAnalyzerQueue) {
```

Comment:
Please, refactor this part where it changes the BPM setting for the analysis. In my implementation you'll see a small change in analyzerBeats in this regard.

This is especially important since one of your changes below removed the restoration of this setting, so rather than restoring the deleted part, better finish the refactor.

Reply:
I just wanted to fix the threading issues and leave the functionality of the existing code unchanged. The same applies to the reading of the configuration property that I didn't scrutinize.

It's good practice to limit changes to a single aspect, although we are already far away from this goal within this PR ;) Of course, if there is an issue it should be fixed eventually!

Reply:
Maybe I don't understand your reply, but I meant that you broke it.
In this commit:
https://github.com/mixxxdj/mixxx/pull/1413/commits/a22df4e45d073cf98ceda49fb7310fe38cd331bb
you removed the call to restore the value from m_iOldBpmEnabled into the settings. 

You can restore that line, or you can take a look at my code and modify the code so that this patch is not needed. (Basically, I implemented it directly into the analyzerbeats).